rfcs/OperationExpressions.md

```diff
@@ -0,0 +1,391 @@
+
+#### Pathing
+
+Following a path from one field to the next could use the `>` character; this is
+already used in Apollo's GraphQL documentation browser and is intuitive for
+navigation. This leaves `.` available and non-ambiguous for referring to fields
+on a type, which is useful when disambiguating references on a union type, for
+instance:
+
+```
+>me>media>Film.duration
+```
+
+might model:
+
+```graphql
+{
+  me {
+    media {
+      ... on Film {
+        duration
+      }
+    }
+  }
+}
+```
+
+#### Operations
+
+The expression `>me>name` would expand to `{ me { name } }`.
+
+If you want to create a mutation or subscription operation, you can prefix the
+path with the operation type (you can do this for queries too, but just like in
+operation documents, the query keyword is optional):
+
+- `mutation>createUser>user>name` expands to
+  `mutation ($input: CreateUserInput!) { createUser(input: $input) { user { name } } }`
+- `subscription>currentUserUpdated>name` expands to
+  `subscription { currentUserUpdated { name } }`
+- `query>me>name` expands to `query { me { name } }`
+
+You may name operations by prefixing with an operation name followed by a colon;
+for example:
+
+- `MyQuery:>me>name` and `MyQuery:query>me>name` expand to
+  `query MyQuery { me { name } }`.
+- `MyMutation:mutation>createUser>name` expands to
+  `mutation MyMutation { createUser { name } }`.
+- `MySubscription:subscription>userCreated>name` expands to
+  `subscription MySubscription { userCreated { name } }`.
+
+#### Fragments
+
+Fragments start with a type name followed by a period: `User.friends>name`
+expands to `... on User { friends { name } }`.
+
+You can name fragments by prefixing with a fragment name and a colon:
+`FriendNames:User.friends>name` expands to
+`fragment FriendNames on User { friends { name } }`.
+
+Other examples:
+
+- `MyFragment:Node.User.fullName:name` expands to
+  `fragment MyFragment on Node { ... on User { fullName: name } }`
+
+- `MyQuery:>allEntities>edges>node>MyNodeFragment:Node.MyUserFragment:User.fullName:name`
+  expands to
+
+  ```graphql
+  query MyQuery {
+    allEntities {
+      edges {
+        node {
+          ...MyNodeFragment
+        }
+      }
+    }
+  }
+
+  fragment MyNodeFragment on Node {
+    ...MyUserFragment
+  }
+
+  fragment MyUserFragment on User {
+    fullName: name
+  }
+  ```
+
+#### Arguments
+
+Arguments use the same syntax as Schema Coordinates; namely parenthesis and a
+colon: `>searchBusinesses(name:)>city`.
+
+We also allow you to reference input objects used in arguments, for example:
+
+`>searchBusinesses(where.size.greaterThan:)>city`
+
+expands to something like:
+
+```graphql
+query($whereSizeGreaterThan: Int) {
+  searchBusinesses(where: { size: { greaterThan: $whereSizeGreaterThan } }) {
+    city
+  }
+}
+```
+
+Further we allow for multiple arguments to be specified, joined with commas:
+
+`>searchBusinesses(where.size.greaterThan:,where.size.lessThan:,where.city.equalTo:)>name`
```

Comment:
I notice we're using a dot separator for input types (`where.size.greaterThan`)

any reason we're opting for . instead of > here?

e.g. this:

`>searchBusinesses(where>size>greaterThan:,where>size>lessThan:,where>city>equalTo:)>name`

(even if the answer is "it looks nicer", then fair enough, just wondering)

Reply:
Good point. Earlier syntax didn’t have >, I had to add it to avoid ambiguity and didn’t think to update the arguments. May be worth adopting as you say to avoid issues with input unions or similar in future.

Reply:
Done :+1: